src/pykeen/lr_schedulers/__init__.py

```diff
@@ -0,0 +1,88 @@
+CyclicLR                     :class:`torch.optim.lr_scheduler.CyclicLR`
+ExponentialLR                :class:`torch.optim.lr_scheduler.ExponentialLR`
+LambdaLR                     :class:`torch.optim.lr_scheduler.LambdaLR`
+MultiplicativeLR             :class:`torch.optim.lr_scheduler.MultiplicativeLR`
+MultiStepLR                  :class:`torch.optim.lr_scheduler.MultiStepLR`
+OneCycleLR                   :class:`torch.optim.lr_scheduler.OneCycleLR`
+StepLR                       :class:`torch.optim.lr_scheduler.StepLR`
+===========================  =============================================================
+
+.. note:: This table can be re-generated with ``pykeen ls lr_schedulers -f rst``
+"""
+
+from typing import Any, Mapping, Type
+
+from class_resolver import Resolver
+from torch.optim.lr_scheduler import (
+    _LRScheduler,
+    CosineAnnealingLR,
+    CosineAnnealingWarmRestarts,
+    CyclicLR,
+    ExponentialLR,
+    LambdaLR,
+    MultiplicativeLR,
+    MultiStepLR,
+    OneCycleLR,
+    StepLR,
+)
+
+__all__ = [
+    '_LRScheduler',
+    'lr_schedulers_hpo_defaults',
+    'lr_scheduler_resolver',
+]
+
+#: The default strategy for optimizing the lr_schedulers' hyper-parameters
+# TODO: Adjust search space to something reasonable
+lr_schedulers_hpo_defaults: Mapping[Type[_LRScheduler], Mapping[str, Any]] = {
+    CosineAnnealingLR: dict(
+        eta_min=dict(type=float, low=0.001, high=0.1, scale='log'),
+        T_max=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+    CosineAnnealingWarmRestarts: dict(
+        T_0=dict(type=float, low=0.001, high=0.1, scale='log'),
+        T_multi=dict(type=float, low=0.001, high=0.1, scale='log'),
+        eta_min=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+    CyclicLR: dict(
+        base_lr=dict(type=float, low=0.001, high=0.1, scale='log'),
+        max_lr=dict(type=float, low=0.001, high=0.1, scale='log'),
+        # TODO: Decide whether all parameters should be available
+    ),
+    ExponentialLR: dict(
+        gamma=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+    LambdaLR: dict(
+        lr_lambda=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+    MultiplicativeLR: dict(
+        lr_lambda=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+    MultiStepLR: dict(
+        gamma=dict(type=float, low=0.001, high=0.1, scale='log'),
+        milestones=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+    OneCycleLR: dict(
+        max_lr=dict(type=float, low=0.001, high=0.1, scale='log'),
+        pct_start=dict(type=float, low=0.001, high=0.1, scale='log'),
+        # TODO: Decide whether all parameters should be available
+    ),
+    StepLR: dict(
+        gamma=dict(type=float, low=0.001, high=0.1, scale='log'),
+        step_size=dict(type=float, low=0.001, high=0.1, scale='log'),
+    ),
+}
+
+lr_scheduler_resolver = Resolver.from_subclasses(
+    _LRScheduler,
+    default=ExponentialLR,
```

Comment:
does this really make sense as a default? Shouldn't it not have one by default?

Reply:
Doesn't really make a difference, since as of now nothing is depending on the default. If you want a lr scheduler this is the simplest one 😄